### PR TITLE
Window function support reorder

### PR DESCRIPTION
**Pull request by HangyuanLiu**

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java [hunk 1]
```diff
@@ -347,10 +354,25 @@ public static WindowOperator standardize(AnalyticExpr analyticExpr) {
                 return sortKey;
             }
         });
-
         return sortedGroups;
     }
 
+    private static boolean isPrefixHyperPartitionSet(List<Object> hyperSet, List<Object> subSet) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:360]: Don't get this method want to do what.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java [hunk 2]
```diff
@@ -347,10 +354,25 @@ public static WindowOperator standardize(AnalyticExpr analyticExpr) {
...
+        if (hyperSet.isEmpty() && subSet.isEmpty()) {
+            return true;
+        }
+
+        List<List<Object>> partitionPrefix = IntStream.rangeClosed(1, hyperSet.size())
+                .mapToObj(i -> hyperSet.subList(0, i)).collect(Collectors.toList());
+        return partitionPrefix.contains(subSet);
+    }
+
+    private static boolean isPrefixHyperSortSet(List<Object> hyperSet, List<Object> subSet) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:370]: Don't get this method want to do what.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java [hunk 3]
```diff
@@ -347,10 +354,25 @@ public static WindowOperator standardize(AnalyticExpr analyticExpr) {
...
     }
 
+    private static boolean isPrefixHyperPartitionSet(List<Object> hyperSet, List<Object> subSet) {
+        if (hyperSet.isEmpty() && subSet.isEmpty()) {
+            return true;
+        }
+
+        List<List<Object>> partitionPrefix = IntStream.rangeClosed(1, hyperSet.size())
+                .mapToObj(i -> hyperSet.subList(0, i)).collect(Collectors.toList());
+        return partitionPrefix.contains(subSet);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:367]: return subSet.stream().allmatch(hyperSet::contains)?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:360]: BTW, partition by don't need column order match

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java [hunk 4]
```diff
@@ -347,10 +354,25 @@ public static WindowOperator standardize(AnalyticExpr analyticExpr) {
...
+    }
+
+    private static boolean isPrefixHyperSortSet(List<Object> hyperSet, List<Object> subSet) {
+        List<List<Object>> partitionPrefix = IntStream.rangeClosed(0, hyperSet.size())
+                .mapToObj(i -> hyperSet.subList(0, i)).collect(Collectors.toList());
+        return partitionPrefix.contains(subSet);
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:373]: for (int i = 0;i < subSet.size(); i++) {
            if (!hyperSet.get(i).equals(subSet.get(i))) {
                return false;
            }

            return true;
        }

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java (hunk 4 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:373]: Same as above

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java (hunk 3 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:367]: No, this logic needs to ensure that the subset is ordered in the hyperset.
eg: HyperSet(A，B，C) , (A，B）is true, (B，A）is false

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:360]: Determine whether a partition satisfies the distribution of another partition, and determine whether to add ExchangeNode

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java (hunk 3 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/WindowTransformer.java:367]: I think it's don't need keep order subset
eg: HyperSet(A，B，C) , (A，B）is true, (B，A）is true too

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**